sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -815,6 +818,77 @@ public void rollbackTransaction(ServiceBusTransactionContext transactionContext)
         asyncClient.rollbackTransaction(transactionContext).block(operationTimeout);
     }
 
+    /**
+     * Deletes up to {@code messageCount} messages from the entity enqueued before {@link OffsetDateTime#now()}.
+     * The actual number of deleted messages may be less if there are fewer eligible messages in the entity.
+     * <p>If the lock for a message is held by a receiver, it will be respected and the message will not be deleted.</p>
+     * <p>You can delete a maximum of 4000 messages in a single API call, this is the current limit determined by
+     * the Service Bus service.</p>
+     *
+     * @param messageCount the desired number of messages to delete.
+     * @return the number of messages deleted.
+     *
+     * @throws IllegalArgumentException when the {@code messageCount} is less than 1 or exceeds the maximum allowed, as
+     * determined by the Service Bus service.
+     */
+    public int deleteMessages(int messageCount) {
+        return deleteMessages(messageCount, new DeleteMessagesOptions());
+    }
+
+    /**
+     * Deletes up to {@code messageCount} messages from the entity. The actual number of deleted messages may be less
+     * if there are fewer eligible messages in the entity.
+     * <p>If the lock for a message is held by a receiver, it will be respected and the message will not be deleted.</p>
+     * <p>You can delete a maximum of 4000 messages in a single API call, this is the current limit determined by
+     * the Service Bus service.</p>
+     *
+     * @param messageCount the desired number of messages to delete.
+     * @param options options used to delete the messages.
+     * @return the number of messages deleted.
+     *
+     * @throws IllegalArgumentException when the {@code messageCount} is less than 1 or exceeds the maximum allowed, as
+     * determined by the Service Bus service.
+     * @throws NullPointerException if {@code options} is null.
+     */
+    public int deleteMessages(int messageCount, DeleteMessagesOptions options) {
+        return asyncClient.deleteMessages(messageCount, options).block(operationTimeout);
```

Comment:
Why do we need timeout here?

we have timeouts per try, i.e. operations should never last forever. This timeout will mess up retry/backoffs mechanism under.

Reply:
Ah, I see that operationTimeout is calculated based on retry options, but I wonder if it's necessary at all. You can never predict how much time it enough since there could be multiple small operations all of which can be retried.
E.g. connection is established after 3 tries, then deletion returned 503 and then connection dropped again.

there is no point(?) in trying to control it on the public surface as long as retry mechanism in lower levels just works.

Reply:
thank you!! great point, yes, let me remove this unnecessary timeout.

Reply:
done